src/main/java/org/apache/sysml/parser/ParseException.java

```diff
@@ -19,74 +19,231 @@
-   * Increment only if the <i>serialized</i> form of the
-   * class changes.
-   */
-  private static final long serialVersionUID = 1L;
-
-  /**
-   * The following constructors are for use by you for whatever
-   * purpose you can think of.  Constructing the exception in this
-   * manner makes the exception behave in the normal way - i.e., as
-   * documented in the class "Throwable".  The fields "errorToken",
-   * "expectedTokenSequences", and "tokenImage" do not contain
-   * relevant information.  The JavaCC generated code does not use
-   * these constructors.
-   */
-
-  public ParseException() {
-    super();
-  }
-
-  /** Constructor with message. */
-  public ParseException(String message) {
-    super(message);
-  }
-  
-  public ParseException(String message, Exception e) {
-	  super(message, e);
-  }
-  /**
-   * This is the last token that has been consumed successfully.  If
-   * this object has been created due to a parse error, the token
-   * following this token will (therefore) be the first error token.
-   */
-  public Token currentToken;
-
-  /**
-   * Each entry in this array is an array of integers.  Each array
-   * of integers represents a sequence of tokens (by their ordinal
-   * values) that is expected at this point of the parse.
-   */
-  public int[][] expectedTokenSequences;
-
-  /**
-   * This is a reference to the "tokenImage" array of the generated
-   * parser within which the parse error occurred.  This array is
-   * defined in the generated ...Constants interface.
-   */
-  public String[] tokenImage;
+public class ParseException extends Exception {
+
+	/**
+	 * The version identifier for this Serializable class. Increment only if the
+	 * <i>serialized</i> form of the class changes.
+	 */
+	private static final long serialVersionUID = 1L;
+
+	/**
+	 * The following constructors are for use by you for whatever purpose you
+	 * can think of. Constructing the exception in this manner makes the
+	 * exception behave in the normal way - i.e., as documented in the class
+	 * "Throwable". The fields "errorToken", "expectedTokenSequences", and
+	 * "tokenImage" do not contain relevant information. The JavaCC generated
+	 * code does not use these constructors.
+	 */
+
+	public ParseException() {
+		super();
+	}
+
+	/** Constructor with message. */
+	public ParseException(String message) {
+		super(message);
+		this.message = message;
+	}
+
+	public ParseException(String message, Exception e) {
+		super(message, e);
+		this.message = message;
+	}
+
+	/**
+	 * This constructor takes a list of parse issues that were generated during
+	 * script parsing and the original DML/PyDML script String.
+	 * 
+	 * @param parseIssues
+	 *            List of parse issues (syntax errors, validation errors, and
+	 *            validation warnings) generated during parsing.
+	 * @param scriptString
+	 *            The DML/PyDML script String.
+	 */
+	public ParseException(List<ParseIssue> parseIssues, String scriptString) {
```

Comment:
I like how you've created separate objects for each of the parse errors. UI components for editing scripts are much easier to write when errors are exposed this way.